specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1358 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+in sbiret.value on success. The host can use this to the number of enumerate individual register
+sets in the vCPU shared-memory state area (also enumerated by the `vcpu_register_set_id` enum).
+The offsets for the state can vary across TSM versions, and they can be determined by calling
+`sbi_tee_host_get_tvm_vcpu_register_set()`.
+
+Note that the VCPU layout is likely to be common across all TVMs, in which case the host can enumerate
+it once. The interface is intended to provide future extensibility to accommodate heterogeneous TVMs
+that may choose to "opt-in" or "opt-out" of specific platform extensions.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_get_tvm_vcpu_num_register_sets_errors]
+.TEE Host Get TVM VCPU Num Register Sets
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code                    | Description
+| SBI_SUCCESS                   | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM         | `tvm_guest_id` was invalid.
+| SBI_ERR_FAILED                | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Get TVM VCPU Register Set (FID #14)
+[source, C]
+-----
+struct sbiret sbi_tee_host_get_tvm_vcpu_register_set(unsigned long tvm_guest_id,
+                                                     unsigned long vcpu_register_set_id);
+-----
+
+The host can use this this interface to discover the shared-memory offset of the VCPU state correspomding
+to the enum values in `vcpu_register_set_id` for `tvm_guest_id`. The `vcpu_register_set_id` parameter must
+be a legal value for the `vcpu_register_set_id` enum.
+
+*Returns* a 32-bit value with the same layout as the `tvm_vcpu_register_set_location` structure in sbiret.value
+on success.
+
+Note that the VCPU layout is likely to be common across all TVMs, in which case the host can enumerate
+it once. The interface is intended to provide future extensibility to accommodate heterogeneous TVMs
+that may choose to "opt-in" or "opt-out" of specific platform extensions.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_get_tvm_vcpu_register_set_errors]
+.TEE Host Get TVM VCPU Register Set
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code                    | Description
+| SBI_SUCCESS                   | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM         | `tvm_guest_id` or `vcpu_register_set_id` was invalid.
+| SBI_ERR_FAILED                | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Create TVM VCPU (FID #15)
+[source, C]
+-----
+struct sbiret sbi_tee_host_create_tvm_vcpu(unsigned long tvm_guest_id,
+                                           unsigned long tvm_vcpu_id,
+                                           unsigned long tvm_vcpu_shared_page_addr);
+-----
```

Comment:
Update this after the changes made in https://github.com/rivosinc/salus/pull/132 ?